tests/test_formparsers.py

```diff
@@ -303,6 +304,16 @@ def test_multipart_request_mixed_files_and_data(tmpdir: Path, test_client_factor
     }
 
 
+def test_multipart_request_large_file(tmpdir: Path, test_client_factory: TestClientFactory) -> None:
+    data = BytesIO(b" " * MultiPartParser.spool_max_size * 2)
+    client = test_client_factory(app)
+    response = client.post(
+        "/",
+        files=[("test_large", data)],
+    )
+    assert response.status_code == 200
```

Comment:
This test passes on master. We want some condition that proves this PR actually solves the issue.

Reply:
@Kludex yeah in it's current state it's not doing a whole lot. Let met try editing the test to use a monkey patched SpooledTemporaryFile to see what thread rollover was called in

Reply:
@Kludex Just pushed the new test! I ended up creating a new class/app to track where rollover was happening. I'm still new to starlette development so please let me know if you have any code-quality suggestions.